src/main/java/com/zendesk/maxwell/row/FieldNameStrategy.java

```diff
@@ -0,0 +1,49 @@
+package com.zendesk.maxwell.row;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * 
+ * @author frankchen Dec 24, 2019 9:08:41 PM
+ */
+public class FieldNameStrategy {
+
+	public final static String UNDERSCORE_TO_CAMEL_CASE = "underscore_to_camelcase";
+
+	// cache converted names
+	private Map<String, String> caches = new ConcurrentHashMap<>();
+
+	private String strategyName;
+
+	public FieldNameStrategy(String strategy) {
+		this.strategyName = strategy;
+	}
+	
+	public String apply(String oldName) {
+		if ( UNDERSCORE_TO_CAMEL_CASE.equals(strategyName)) 
+			return caches.computeIfAbsent(oldName, k -> underscore2Camel(k));
```

Comment:
wondering how necessary the underscore-to-camel-case cache is.  My guess is that if you measured it, the cost of `String#hash` would be around the same as the cost of `underscore2Camel`.  Or close enough.

Reply:
also please *do* cache the result of `UNDERSCORE_TO_CAMEL_CASE.equals(strategyName)` in the constructor so you only have to do that string comparison once.

Reply:
> wondering how necessary the underscore-to-camel-case cache is. My guess is that if you measured it, the cost of `String#hash` would be around the same as the cost of `underscore2Camel`. Or close enough.

time consumption maybe the same, but cache prevents lots of temporary objects allocation and gc

Reply:
> also please _do_ cache the result of `UNDERSCORE_TO_CAMEL_CASE.equals(strategyName)` in the constructor so you only have to do that string comparison once.

i've fixed it, please check it